engine/flink/kafka-util/src/main/scala/pl/touk/nussknacker/engine/kafka/generic/DelayedFlinkKafkaConsumer.scala

```diff
@@ -87,8 +91,10 @@ class DelayedKafkaFetcher[T](sourceContext: SourceFunction.SourceContext[T],
     var maxEventTimestamp = 0L
     records.forEach(new Consumer[T]{
       override def accept(r: T): Unit = {
-        val recordTimestamp = partitionState.extractTimestamp(r, kafkaEventTimestamp)
-        if(recordTimestamp > maxEventTimestamp){
+        val recordTimestamp = timestampAssigner
+          .flatMap(_.extractTimestamp(r, kafkaEventTimestamp))
+          .getOrElse(partitionState.extractTimestamp(r, kafkaEventTimestamp))
```

Comment:
Please add comment that partitionState.extractTimestamp works correctly only for brand-new timestamp handlers (for legacy not)

Reply:
Comment added.